resolver/resolver.go

```diff
@@ -29,9 +29,7 @@ type Resolver struct {
 	rs      *records.RecordGenerator
 	rsLock  sync.RWMutex
 	rng     *rand.Rand
-
-	// pluggable external DNS resolution, mainly for unit testing
-	extResolver exchanger.Exchanger
+	exchanger.Forwarder
```

Comment:
Defining a func on a func that calls itself is usually done to implement an
interface. Without the interface it adds unnecessary complexity to the API
(unless, of course, it adds some business/app logic value -- which this
impl doesn't).

Forwarder is already a func type. So something like `resolver.fwd(...)`
reads just as easily as what you have now, makes Forwarder field private,
and keeps the exchanger API simpler. If you don't like `fwd` for a field
name then name it `forward` and then the code will read
`resolver.forward(...)` which basically maintains the readability + the
other pros I mentioned.


On Wed, Oct 7, 2015 at 5:28 PM, Tomás Senart <notifications@github.com>
wrote:

> In resolver/resolver.go
> <https://github.com/mesosphere/mesos-dns/pull/307#discussion_r41450040>:
>
> > @@ -29,9 +29,7 @@ type Resolver struct {
> >  	rs      *records.RecordGenerator
> >  	rsLock  sync.RWMutex
> >  	rng     *rand.Rand
> > -
> > -	// pluggable external DNS resolution, mainly for unit testing
> > -	extResolver exchanger.Exchanger
> > +	exchanger.Forwarder
>
> I'm pro making this private, but you have to convince me of the
> *confusion* associated with the Forward method. :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/307/files#r41450040>.
>